crypto/asn1/a_d2i_fp.c

```diff
@@ -138,15 +140,18 @@ int asn1_d2i_read_bio(BIO *in, BUF_MEM **pb)
 
         p = (unsigned char *)&(b->data[off]);
         q = p;
-        inf = ASN1_get_object(&q, &slen, &tag, &xclass, len - off);
+        diff = len - off;
+        if (diff == 0)
+            goto err;
+        inf = ASN1_get_object(&q, &slen, &tag, &xclass, diff);
         if (inf & 0x80) {
             unsigned long e;
 
             e = ERR_GET_REASON(ERR_peek_error());
             if (e != ASN1_R_TOO_LONG)
                 goto err;
             else
```

Comment:
else after goto is bad style because you're just falling through to the next lines anyway

Reply:
Right - so I've cleaned this up as well.

(There would be also deeper things to clean up, such as the use of needlessly non-conservative `ERR_clear_error()`, but that's out of scope of this PR.)